clang/lib/Lex/Preprocessor.cpp

```diff
@@ -878,6 +878,25 @@ bool Preprocessor::HandleIdentifier(Token &Identifier) {
   return true;
 }
 
+void Preprocessor::saveCheckPoint(const char *P) {
+  static constexpr ptrdiff_t Limit = 1024 * 8;
+  if (CheckPoints.empty()) {
+    CheckPoints.push_back(P);
+    return;
+  }
+
+  const char *Cur = CheckPoints.back();
+  if (Cur == P)
+    return;
```

Comment:
can this actually happen? Should it be an assert instead?

Reply:
IIRC yes, this did happen.

Reply:
Thinking about this some more, I think this was only happening when I tested with `Limit = 1`. I'll re-check.